lxd/resources/pci.go

```diff
@@ -0,0 +1,106 @@
+package resources
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/jaypipes/pcidb"
+	"github.com/pkg/errors"
+
+	"github.com/lxc/lxd/shared/api"
+)
+
+// GetPCI returns a filled api.ResourcesPCI struct ready for use by LXD
+func GetPCI() (*api.ResourcesPCI, error) {
+	pci := api.ResourcesPCI{}
+
+	// Load PCI database
+	pciDB, err := pcidb.New()
+	if err != nil {
+		pciDB = nil
+	}
+
+	// List all PCi devices
+	entries, err := ioutil.ReadDir(sysBusPci)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to list %q", sysBusPci)
+	}
+
+	for _, entry := range entries {
```

Comment:
I'd follow the same patterns as the PCI iteration loop in `network.go` and use:
```
entryName := entry.Name()
devicePath := filepath.Join(sysBusPci, entryName)
```

Makes copy/pasting and spotting bugs much easier if the structure and naming matches.